jerry-main/main-unix-snapshot.c

```diff
@@ -389,36 +374,199 @@ process_generate (cli_state_t *cli_state_p, /**< cli state */
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: at least one input file must be specified.\n");
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+  jerry_init (JERRY_INIT_EMPTY);
+
+  size_t lit_buf_sz = 0;
+  if (number_of_files == 1)
+  {
+    lit_buf_sz = jerry_get_literals_from_snapshot (snapshot_buffers[0],
+                                                   snapshot_buffer_sizes[0],
+                                                   literal_buffer,
+                                                   JERRY_BUFFER_SIZE,
+                                                   is_c_format);
+  }
+  else
+  {
+    /* The input contains more than one input snapshot file, so we must merge them first. */
+    const char *error_p = NULL;
+    size_t merged_snapshot_size = jerry_merge_snapshots (snapshot_buffers,
+                                                         snapshot_buffer_sizes,
+                                                         number_of_files,
+                                                         output_buffer,
+                                                         JERRY_BUFFER_SIZE,
+                                                         &error_p);
+
+    if (merged_snapshot_size == 0)
     {
-      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: Unable to write literal file: '%s'\n", literals_file_name_p);
+      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", error_p);
+      jerry_cleanup ();
       return JERRY_STANDALONE_EXIT_CODE_FAIL;
     }
 
-    fwrite (output_buffer, sizeof (uint8_t), literal_buffer_size, literal_file_p);
-    fclose (literal_file_p);
+    printf ("Successfully merged the input snapshots (%lu bytes).\n", merged_snapshot_size);
 
-    printf ("Created literal file: '%s' (%lu bytes)\n", literals_file_name_p, (unsigned long) literal_buffer_size);
+    lit_buf_sz = jerry_get_literals_from_snapshot (output_buffer,
+                                                   merged_snapshot_size,
+                                                   literal_buffer,
+                                                   JERRY_BUFFER_SIZE,
+                                                   is_c_format);
   }
 
-  return 0;
-} /* process_generate */
+  if (lit_buf_sz == 0)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR,
+                    "Error: Literal saving failed! No literals were found in the input snapshot(s).\n");
+    jerry_cleanup ();
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+  if (literals_file_name_p == NULL)
+  {
+    literals_file_name_p = is_c_format ? "literals.h" : "literals.list";
+  }
+
+  FILE *file_p = fopen (literals_file_name_p, "wb");
+
+  if (file_p != NULL)
+  {
+    fwrite (literal_buffer, sizeof (uint8_t), lit_buf_sz, file_p);
+    fclose (file_p);
+  }
+  else
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: cannot open file: '%s'\n", literals_file_name_p);
+    jerry_cleanup ();
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
```

Comment:
the `if (...) { log and return } ...` (without an `else` branch) style is used all over the code, and could also be used here: `if (file_p == NULL) { bail out }`

Reply:
fixed